exercises/concept/rpn-calculator-inspection/test/rpn_calculator_inspection_test.exs

```diff
@@ -0,0 +1,334 @@
+               test_data,
+               check_results_so_far
+             ) ==
+               expected_result
+
+      assert Keyword.get(Process.info(self()), :trap_exit) == false
+      Process.flag(:trap_exit, true)
+
+      send(caller_process_pid, {:EXIT, caller_process_pid, :normal})
+
+      assert RPNCalculatorInspection.await_reliability_check_result(
+               test_data,
+               check_results_so_far
+             ) ==
+               expected_result
+
+      assert Keyword.get(Process.info(self()), :trap_exit) == true
+    end
+  end
+
+  describe "reliability_check" do
+    @tag :pending
+    test "returns an empty map when input list empty" do
+      inputs = []
+      calculator = &RPNCalculator.unsafe_division/1
+      outputs = %{}
+      assert RPNCalculatorInspection.reliability_check(calculator, inputs) == outputs
+    end
+
+    @tag :pending
+    test "returns a map with inputs as keys and :ok as values" do
+      inputs = ["4 2 /", "8 2 /", "6 3 /"]
+      calculator = &RPNCalculator.unsafe_division/1
+
+      outputs = %{
+        "4 2 /" => :ok,
+        "8 2 /" => :ok,
+        "6 3 /" => :ok
+      }
+
+      assert RPNCalculatorInspection.reliability_check(calculator, inputs) == outputs
+    end
+
+    @tag :pending
+    test "returns a map when input list has 1000 elements" do
+      inputs = Enum.map(1..1000, &"#{2 * &1} #{&1} /")
+      calculator = &RPNCalculator.unsafe_division/1
+      outputs = 1..1000 |> Enum.map(&{"#{2 * &1} #{&1} /", :ok}) |> Enum.into(%{})
+
+      assert RPNCalculatorInspection.reliability_check(calculator, inputs) == outputs
+    end
+
+    @tag :pending
+    test "returns a map when input list has 1000 elements and the calculator takes 50ms for each calculation" do
+      inputs = Enum.map(1..1000, &"#{2 * &1} #{&1} /")
+      calculator = fn input -> :timer.sleep(50) && RPNCalculator.unsafe_division(input) end
+      outputs = 1..1000 |> Enum.map(&{"#{2 * &1} #{&1} /", :ok}) |> Enum.into(%{})
+
+      assert RPNCalculatorInspection.reliability_check(calculator, inputs) == outputs
+    end
+
+    @tag :pending
+    test "returns :error values for inputs that cause the calculator to crash" do
+      inputs = ["3 0 /", "22 11 /", "4 0 /"]
+      calculator = &RPNCalculator.unsafe_division/1
+
+      outputs = %{
+        "3 0 /" => :error,
+        "22 11 /" => :ok,
+        "4 0 /" => :error
+      }
+
+      assert RPNCalculatorInspection.reliability_check(calculator, inputs) == outputs
+    end
+
+    #    @tag :pending
+    #    test "returns a map when input list has 1000 elements and all of them crash" do
+    #      inputs = Enum.map(1..1000, &"#{2 * &1} 0 /")
+    #      calculator = &RPNCalculator.unsafe_division/1
+    #      outputs = 1..1000 |> Enum.map(&{"#{2 * &1} 0 /", :error}) |> Enum.into(%{})
+    #
+    #      assert RPNCalculatorInspection.reliability_check(calculator, inputs) == outputs
+    #    end
```

Comment:
I have a problem with this test, and with the `"returns :error values for inputs that cause the calculator to crash"` test. They produce a lot of output that I don't know how to mute. For each linked process spawned in `reliability_check` that crashes, there is an error log in the test output like this:

```
17:28:30.824 [error] Process #PID<0.4204.0> raised an exception
** (ArithmeticError) bad argument in arithmetic expression
    test/rpn_calculator_inspection_test.exs:7: RPNCalculatorInspectionTest.RPNCalculator.unsafe_division/1
```

I tried to mute it with `capture_io` and `caputure_log` from ExUnit, but that doesn't work (probably this output is produced from a different process). I'm not sure how to get rid of it, but I think we have to because:
- it's expected, it doesn't mean an error from the user's perspective
- it will mess up real useful error output from the test if the student actually struggled with this test
- because I don't know which process produces it, it might mean that this error output is asynchronous relative to the test and might "leak" into the output of other tests

Reply:
This is being routed through the logger interface. You could try to suppres logging events during tests, but I am not sure how that might affect things

Reply:
`Logger.configure(level: :none)` in the test setup helps! It's not ideal because it will prevent the student from using the logger to debug the tests, but the debugging technique we recommend is `IO.inspect` anyway, and that still works ;).

Reply:
Awesome! I also think this is quite reasonable to do here.  As you said, we recommend debugging using stdout with `IO.puts` and `IO.inspect` and setting up a logger might be outside of the scope of the exercism track.